snowplow-tracker/src/main/java/com/snowplowanalytics/core/tracker/TrackerWebViewInterfaceV2.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2015-present Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.core.tracker
+
+import android.webkit.JavascriptInterface
+import com.snowplowanalytics.core.event.WebViewReader
+import com.snowplowanalytics.core.tracker.Logger.e
+import com.snowplowanalytics.core.utils.JsonUtils.jsonToMap
+import com.snowplowanalytics.snowplow.Snowplow.defaultTracker
+import com.snowplowanalytics.snowplow.Snowplow.getTracker
+import com.snowplowanalytics.snowplow.event.*
+import com.snowplowanalytics.snowplow.payload.SelfDescribingJson
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.*
+
+/**
+ * JavaScript interface used to provide an API for tracking events from WebViews.
+ */
+class TrackerWebViewInterfaceV2 {
+    @JavascriptInterface
+    @Throws(JSONException::class)
+    fun trackWebViewEvent(
```

Comment:
I'm slightly worried about this long list of parameters for 2 reasons:

1. It might cause breaking changes to introduce new parameters. What if we decide to also track `domain_sessionid` in the future? Will that make new WebView trackers tracking that extra property incompatible with older mobile trackers with WebViewInterface not having that extra parameter?
2. It's not a very nice API – I mean it's easy to make a mistake on the JS side and mess up the order of parameters being sent (there aren't named function params in JS).

My suggestion would be to consolidate the atomic properties into a JSON string and maybe have the following parameters:

1. selfDescribingEventData: String?
2. entities: String?
3. atomicProperties: String?

What do you think?

Reply:
no named function params in JS 🫠